include/roboteam_ai/control/BBTrajectories/BBTrajectory2DAsync.h

```diff
@@ -0,0 +1,33 @@
+//
+// Created by rolf on 18-01-20.
+//
+
+#ifndef RTT_BBTRAJECTORY2DASYNC_H
+#define RTT_BBTRAJECTORY2DASYNC_H
+#include "BBTrajectory2D.h"
+namespace rtt {
+template<class num>
+class BBTrajectory2DAsync {
+    private:
+        BBTrajectory1D<num> x;
+        BBTrajectory1D<num> y;
+        num rotation;
+        Vector2 startPosition;
+    public:
+        BBTrajectory2DAsync() = default;
+        BBTrajectory2DAsync(const Vector2 &initialPos, const Vector2 &initialVel, const Vector2 &finalPos,
+                num maxVel, num maxAcc, const LineSegment &line);
+        void generateTrajectory( const Vector2 &initialVel, const Vector2 &finalPos,
+                num maxVel, num maxAcc, num alpha) noexcept;
+        void generateAsyncTrajectory(const Vector2 &initialPos, const Vector2 &initialVel, const Vector2 &finalPos,
+                num maxVel, num maxAcc, const LineSegment &line) noexcept;
+        Vector2 getPosition(num t) const;
+        Vector2 getVelocity(num t) const;
+        Vector2 getAcceleration(num t) const;
+        std::vector<Vector2> visCurve() const;
+
+};
```

Comment:
How is this class async?

Reply:
Ah sorry the naming might be a bit confusing indeed. It's 'async'  in the sense that in the other class we synchronize the trajectories to take the same amount of time, and here we don't. I might want to give it another name to prevent confusion, this has nothing to do with threading/concurrency.